### ISSUE TITLE
Dr. Memory segfaults in Docker container

### ISSUE THREAD

**Issue by mrigger** [DynamoRIO/drmemory#2133]
As explained below, Dr. Memory aborts with a segfault when being used in a Docker container.

Here is the `Dockerfile` to produce a minimal failing example:
```
FROM ubuntu:18.04
RUN apt-get update && apt-get install -y --no-install-recommends wget software-properties-common gpg-agent
RUN wget -O - "https://github.com/DynamoRIO/drmemory/releases/download/release_2.0.1/DrMemory-Linux-2.0.1-2.tar.gz" | tar -xz -C /
ENTRYPOINT ["DrMemory-Linux-2.0.1-2/bin64/drmemory", "--", "wget", "--help"] FOREGROUND
```
I used `docker build - < Dockerfile` to build the container and `docker run` to run it. Unexpectedly, Dr. Memory fails during execution:

```
~~Dr.M~~ Dr. Memory version 2.0.1
<Application /usr/bin/wget (1).  Dr. Memory internal crash at PC 0x00007f86a71e2244.  Please report this at http://drmemory.org/issues.  Program aborted.
Received SIGSEGV at pc 0x00007f86a71e2244 in thread 1
Base: 0x00007f86a7107000
Registers:eax=0x0000000000000000 ebx=0x000000004cf785b0 ecx=0x0000000000000000 edx=0x0000000000000080
	esi=0x0000000000000000 edi=0x0000000000008f01 esp=0x000000004cde6ba0 ebp=0x0000000000000000
	r8 =0x0000000000000000 r9 =0x000000000000006a r10=0x0000000000000001 r11=0x0000000000000006
	r12=0x0000000000000000 r13=0x0000000000000000 r14=0x000000000000000e r15=0x00007f86a7485298
	eflags=0x0000000000010282
2.0.1-2-(Aug 28 2018 13:50:13)0
-no_dynamic_options -disasm_mask 8 -logdir '/DrMemory-Linux-2.0.1-2/drmemory/logs/dynamorio' -client_lib '/DrMemory-Linux-2.0.1-2/bin64/release/libdrmemorylib.so;0;-logdir `/DrMemory-Linux-2.0.1-2/drmemory/logs` -symcache_dir `/DrMemory-Linux-2.0.1-2/drmemory/logs/symcache` -resfile 1 ' -code_api -stack_size 56K -signal_
0x00007f86a7240918 0x3025783020435020>
```

When running in light mode (`-light`), Dr. Memory executes successfully. I used Docker version `18.03.1-ce`.

**Comment by mrigger** [comment]
Are there any updates on this?

**Comment by derekbruening** [comment]
Does `-debug -dr_debug` report a different error?  Same problem in 2.1.0 version?

**Comment by mrigger** [comment]
Yes, using `-debug -dr_debug` results in the following error (see [here](https://gist.githubusercontent.com/mrigger/0a879fc6adc519a484aaec50870e8d32/raw/30de14fa96df152f32f77f5c5d4730a24e5fdc1d/gistfile1.txt) for the full output):

```
<Application /usr/bin/wget (1) DynamoRIO usage error : encode_cti error: target beyond 8-bit reach>
<Usage error: encode_cti error: target beyond 8-bit reach (/drmemory_package/dynamorio/core/arch/x86/encode.c, line 2174)
2.0.1-2-(Aug 28 2018 13:50:51)0
```

It seems that the issue does not appear in the latest version of Dr. Memory (2.1.0), so feel free to close this GitHub issue. Just out of interest, do you know what caused or could have caused the crash?

**Comment by derekbruening** [comment]
Yes, there were several short jumps used in places that may not reach in 64-bit mode. 
 It may have been fixed by 93fa04463985f7b6ba49abeb6b487448c72b5c70 or 62aa3566c946c957d7191453c34a3d140871d104.  Thank you for the information; closing.
